nvflare/job_config/fed_job.py

```diff
@@ -0,0 +1,279 @@
+        self.job.set_site_app(target, app_name)
+
+    def _run_deploy(self):
+        if not self._deployed:
+            for target in self._deploy_map:
+                self._deploy(self._deploy_map[target], target)
+
+            self._deployed = True
+
+    def export_job(self, job_root):
+        self._run_deploy()
+        self.job.generate_job_config(job_root)
+
+    def simulator_run(self, workspace, threads: int = None):
+        self._run_deploy()
+
+        n_clients = len(self.clients)
+        if threads is None:
+            threads = n_clients
+
+        self.job.simulator_run(
+            workspace,
+            clients=",".join(self.clients),
+            n_clients=n_clients,
+            threads=threads,
+            gpu=",".join([self._gpus[client] for client in self._gpus.keys()]),
+        )
+
+
+class ExecutorApp(FedApp):
+    def __init__(self, external_scripts: List = None):
+        """Wrapper around `ClientAppConfig`.
+
+        Args:
+            external_scripts: List of external scripts that need to be deployed to the client. Defaults to None.
+        """
+        super().__init__()
+        self.external_scripts = external_scripts
+        self._create_client_app()
+
+    def add_executor(self, executor, tasks=None):
+        if tasks is None:
+            tasks = ["*"]  # Add executor for any task by default
+        self.app.add_executor(tasks, executor)
+
+    def _create_client_app(self):
+        self.app = ClientAppConfig()
+
+        component = FilePipe(  # TODO: support CellPipe, causes type error for passing secure_mode = "{SECURE_MODE}"
+            mode=Mode.PASSIVE,
+            root_path="{WORKSPACE}/{JOB_ID}/{SITE_NAME}",
+        )
+        self.app.add_component("metrics_pipe", component)
+
+        component = MetricRelay(pipe_id="metrics_pipe", event_type="fed.analytix_log_stats", read_interval=0.1)
+        self.app.add_component("metric_relay", component)
+
+        component = ExternalConfigurator(component_ids=["metric_relay"])
+        self.app.add_component("config_preparer", component)
+
+        if self.external_scripts is not None:
+            for _script in self.external_scripts:
+                self.app.add_ext_script(_script)
+
+
+class ControllerApp(FedApp):
+    """Wrapper around `ServerAppConfig`.
+
+    Args:
+    """
+
+    def __init__(self, key_metric="accuracy"):
+        super().__init__()
+        self.key_metric = key_metric
+        self._create_server_app()
+
+    def add_controller(self, controller, id=None):
+        if id is None:
+            id = "controller"
+        self.app.add_workflow(self._check_id(id), controller)
+
+    def _create_server_app(self):
+        self.app: ServerAppConfig = ServerAppConfig()
+
+        component = ValidationJsonGenerator()
+        self.app.add_component("json_generator", component)
+
+        if self.key_metric:
+            component = IntimeModelSelector(key_metric=self.key_metric)
```

Comment:
It can be configured at the FedJob level. See [here](https://github.com/holgerroth/NVFlare/blob/3b1d9e2fc6bc9447900de1fa5278a122fd1880af/nvflare/job_config/fed_job.py#L103).

Reply:
Right. The user can send the negative loss or use `negate_key_metric=True` in the selector. In the second case, that argument needs to be exposed. I created an issue https://github.com/NVIDIA/NVFlare/issues/2585